src/pages/ServiceDetails/ServiceDetailsPage.tsx

```diff
@@ -39,6 +73,113 @@ class ServiceDetails extends React.Component<RouteComponentProps<ServiceId>, Ser
     };
     NamespaceFilterSelected.setSelected([activeFilter]);
   };
+
+  parseServiceDetailsInfo = data => {
+    let parsed: ServiceDetailsInfo = {
+      labels: data.labels,
+      name: data.name,
+      created_at: data.created_at,
+      resource_version: data.resource_version,
+      type: data.type,
+      ports: data.ports,
+      endpoints: data.endpoints,
+      istio_sidecar: hasIstioSidecar(data.deployments),
+      deployments: data.deployments,
+      dependencies: data.dependencies,
+      routeRules: data.route_rules,
+      destinationPolicies: data.destination_policies,
+      virtualServices: data.virtual_services,
+      destinationRules: data.destination_rules,
+      ip: data.ip,
+      health: data.health
+    };
+    return parsed;
+  };
+
+  validateParams(parsed: ParsedSearch): boolean {
+    if (!parsed.type || !parsed.name) {
+      return false;
+    }
+    // Check we have the right parameter
+    let validateTypes = ['routerule', 'destinationpolicy', 'virtualservice', 'destinationrule'];
+    if (parsed.type && validateTypes.indexOf(parsed.type) < 0) {
+      return false;
+    }
+    if (parsed.type === 'routerule' && this.state.serviceDetailsInfo.routeRules) {
```

Comment:
Is there a way to avoid repeating this if-else and for statement? In typescript can we do not-typed methods?
Thinking about a method `function find(untypedIstioObjectList, name) boolean`.

Reply:
Yes, but then it means to not use types and treat those as plain javascript.
I considered that, but I prefer to maintain the strong types here as there are different concepts.

Reply:
I was thinking about a switch case but there is a different loop in each condition so...